File: Design-Patterns/Bridge/CSharp/Bridge/Spells/IceSpell.cs
using System;

namespace Bridge.Spells
{
    public class IceSpell : Spell
    {
        public IceSpell(int power, int speed, int cost) : base(power, speed, cost)
        {

        }

        public override void Cast()
        {
            Console.WriteLine($"Casts a ice spell with power {Power} and speed {Speed}. Freezes the enemy 3 seconds.");
        }
    }
}

File: Design-Patterns/Bridge/Cpp/ice_spell.cpp
#include <iostream>
#include <string>
#include "spell.h"

ice_spell::ice_spell(int power, int speed, int cost) : spell(power, speed, cost)
{

}

void ice_spell::cast()
{
	std::cout << "Casts a ice spell with power " + std::to_string(power) + " and speed " + std::to_string(speed) + ". Freezes the enemy 3 seconds." << std::endl;
}